src/components/header.tsx

```diff
@@ -1,42 +1,33 @@
 import { Link } from "gatsby"
-import PropTypes from "prop-types"
 import React from "react"
+import { Box, Text } from "rebass"
 
-const Header = ({ siteTitle }) => (
-  <header
-    style={{
-      background: `rebeccapurple`,
-      marginBottom: `1.45rem`,
-    }}
-  >
-    <div
+const Header: React.FunctionComponent<{ siteTitle: string }> = ({
+  siteTitle,
+}) => (
+  <Box bg="rebeccapurple" mb="1.45rem">
+    <Box
+      my={0}
+      mx="auto"
+      p="1.45rem 1.0875rem"
       style={{
         margin: `0 auto`,
-        maxWidth: 960,
         padding: `1.45rem 1.0875rem`,
       }}
     >
-      <h1 style={{ margin: 0 }}>
-        <Link
-          to="/"
-          style={{
-            color: `white`,
-            textDecoration: `none`,
-          }}
-        >
+      <Link
+        to="/"
+        style={{
+          color: `white`,
+          textDecoration: `none`,
+        }}
+      >
+        <Text as="h1" fontFamily="sans" style={{ margin: 0 }}>
```

Comment:
Rebass provides root components that hook into your theme, removing the need to use css for most layout concerns. In this case `Text` will render as an `h1` and receive `theme.fonts['sans']` as its css `font-family`. If I'd wanted I could have done `m={0}` to select the 0th element in my rebass' default `theme.space` array (which happens to be 0). The docs for `rebass` & `styled-system` cover this in much more detail than my PR could.